server/src/main/java/org/elasticsearch/cluster/routing/allocation/RoutingAllocation.java

```diff
@@ -451,18 +454,24 @@ public RoutingAllocation immutableClone() {
         );
     }
 
-    public RoutingAllocation mutableCloneForSimulation() {
+    public RoutingAllocation mutableCloneForSimulation(int targetRelocatingShardsDuringSimulation) {
         return new RoutingAllocation(
             deciders,
             clusterState.mutableRoutingNodes(),
             clusterState,
             clusterInfo,
             shardSizeInfo,
             currentNanoTime,
-            true
+            true,
+            targetRelocatingShardsDuringSimulation
         );
     }
 
+    public boolean hasTargetRelocatingShards() {
+        assert isSimulating : "should only be called while simulating";
+        return isSimulating && routingNodes().getRelocatingShardCount() >= targetRelocatingShardsDuringSimulation;
```

Comment:
Was there a reason you wanted the escape hatch to not return early?

Reply:
Similarly, the change is not relevant anymore.